db/config.cc

```diff
@@ -911,6 +911,7 @@ db::config::config(std::shared_ptr<db::extensions> exts)
     , cdc_dont_rewrite_streams(this, "cdc_dont_rewrite_streams", value_status::Used, false,
             "Disable rewriting streams from cdc_streams_descriptions to cdc_streams_descriptions_v2. Should not be necessary, but the procedure is expensive and prone to failures; this config option is left as a backdoor in case some user requires manual intervention.")
     , strict_allow_filtering(this, "strict_allow_filtering", liveness::LiveUpdate, value_status::Used, strict_allow_filtering_default(), "Match Cassandra in requiring ALLOW FILTERING on slow queries. Can be true, false, or warn. When false, Scylla accepts some slow queries even without ALLOW FILTERING that Cassandra rejects. Warn is same as false, but with warning.")
+    , strict_is_not_null_in_views(this, "strict_is_not_null_in_views", liveness::LiveUpdate, value_status::Used,db::tri_mode_restriction_t::mode::WARN, "In materialized views the IS NOT NULL restrictions works only on the view's primary key columns. On other columns the restriction isn't respected. Can be true, false, or warn. When false, Scylla accepts IS NOT NULL restrictions on regular columns, but they're silently ignored. It's useful for backwards compatibility. When set to true, IS NOT NULL is allowed only on the view's primary key columns, trying to use it on regular columns will cause an error. Warn is same as false, but with a warning.")
```

Comment:
You mention scylla.yaml in the commit message, but I don't see it in this patch.
Your idea of putting "true" in the yaml and "warn" in the code is an interesting one, with a lot of appeal. The flip side is that a new cluster with an **existing** application will not work. Although perhaps this is good enough (when moving an existing application to a new cluster, presumably some period of instability and changes is to be expected).

Reply:
Agree that a new cluster is better strict. Note that the failure will be very quick (as soon as the schema is created).

Reply:
> You mention scylla.yaml in the commit message, but I don't see it in this patch.

At first I based my changes on the ones for `strict_allow_filtering`, and there was no `strict_allow_filtering` in `scylla.yaml`, so I wasn't sure if I should add it, maybe it gets added automatically by some script (?).

I added my flag to `scylla.yaml` now. I assume that `strict_allow_filtering` should also be there, but we forgot to add it there. I will open an issue about it.

Reply:
Opened https://github.com/scylladb/scylladb/issues/14164